src/SprykerEco/Zed/Unzer/Business/Payment/Resolver/UnzerMarketplacePaymentUnzerCredentialsResolver.php

```diff
@@ -0,0 +1,114 @@
+        $this->unzerPaymentMethodsAdapter = $unzerPaymentMethodsAdapter;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\UnzerMarketplacePaymentCredentialsResolverCriteriaTransfer $unzerMarketplacePaymentCredentialsResolverCriteriaTransfer
+     *
+     * @return \Generated\Shared\Transfer\UnzerCredentialsTransfer
+     */
+    public function resolveMarketplacePaymentUnzerCredentials(
+        UnzerMarketplacePaymentCredentialsResolverCriteriaTransfer $unzerMarketplacePaymentCredentialsResolverCriteriaTransfer
+    ): UnzerCredentialsTransfer {
+        $quoteTransfer = $unzerMarketplacePaymentCredentialsResolverCriteriaTransfer->getQuoteOrFail();
+        $paymentMethodKey = $unzerMarketplacePaymentCredentialsResolverCriteriaTransfer->getPaymentMethodKeyOrFail();
+
+        if ($this->hasMarketplaceMerchantUnzerCredentials($quoteTransfer)) {
+            $unzerCredentialsTransfer = $this->getMainMarketplaceUnzerCredentials($quoteTransfer);
+            $paymentMethodsTransfer = $this->unzerPaymentMethodsAdapter->getPaymentMethods($unzerCredentialsTransfer->getUnzerKeypairOrFail());
+
+            if ($this->isMainMarketplacePaymentMethodAvailable($paymentMethodsTransfer, $paymentMethodKey)) {
+                return $unzerCredentialsTransfer;
+            }
+        }
+
+        return $quoteTransfer->getUnzerCredentialsOrFail();
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PaymentMethodsTransfer $paymentMethodsTransfer
+     * @param string $paymentMethodKey
+     *
+     * @return bool
+     */
+    protected function isMainMarketplacePaymentMethodAvailable(PaymentMethodsTransfer $paymentMethodsTransfer, string $paymentMethodKey): bool
+    {
+        foreach ($paymentMethodsTransfer->getMethods() as $paymentMethodTransfer) {
+            if ($paymentMethodTransfer->getPaymentMethodKey() === $paymentMethodKey) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return bool
+     */
+    protected function hasMarketplaceMerchantUnzerCredentials(QuoteTransfer $quoteTransfer): bool
+    {
+        return !in_array(
+            $quoteTransfer->getUnzerCredentialsOrFail()->getTypeOrFail(),
+            [
+                UnzerConstants::UNZER_CREDENTIALS_TYPE_STANDARD,
+                UnzerConstants::UNZER_CREDENTIALS_TYPE_MAIN_MARKETPLACE,
+            ],
+            true,
+        );
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return \Generated\Shared\Transfer\UnzerCredentialsTransfer
+     */
+    protected function getMainMarketplaceUnzerCredentials(QuoteTransfer $quoteTransfer): UnzerCredentialsTransfer
+    {
+        $unzerCredentialsConditionsTransfer = (new UnzerCredentialsConditionsTransfer())
+            ->addStoreName($quoteTransfer->getStoreOrFail()->getNameOrFail())
```

Comment:
it affects several facade methods
required properties are missing at specs

Reply:
Looks fine for me, do you mean something specific?

Reply:
`performPreSaveOrderStack` facade method seems to be affected by these changes

Reply:
Fixed.